### DISCUSSION TITLE
Required Subsonic API version inconsistency

### DISCUSSION THREAD

**Discussion by spl0k** [opensubsonic/open-subsonic-api#31] [General]
Hello.

I noticed two pages are giving somewhat contradictory requirements.
The [OpenSubsonic changes](https://opensubsonic.netlify.app/docs/opensubsonic-changes/) page tells that a server should respond with an 41 error if it doesn't support the "token authentication" while advertising as >1.13.0. This supposes that a server adhering to the OpenSubsonic standard can advertise itself as supporting any Subsonic API version.
Meanwhile, the [Subsonic API versions](https://opensubsonic.netlify.app/docs/subsonic-versions/) page requires servers to support at least 1.14.0.

What is the actual requirement? Why should there be a requirement on the supported Subsonic API version in the first place?

**Comment by Tolriq** [comment]
> OpenSubsonic servers should support at least 1.14.0. It is not required to support version 1.16.1, but this is still highly recommended.

It's clearly stated **should**. Because it's the first version that have most of the feature a modern client would need, it's nowhere mandatory.

The Error 41 thing is because many servers currently advertise API 1.16.1 but do not support the new auth. (Mostly because it requires to store the password in clear and they refuse, something I can understand). 
So this rule is added to allow servers to be at a version 1.16.1 (Or any version > 1.13.0) but to not support this Subsonic "new authentification". I think most servers in this org now properly return error 41 or support the auth. It's just added as a rules for future servers. And so that clients can properly detect this and fallback to default auth.

The goal being that we can work out the new authentification solution so that no one actually need to use that 1.13+ authentification system.

**Reply by spl0k** [reply]
Thank you for the clarification. I tend to read "you should" as "you must" rather than "you are strongly encouraged to".

Still I don't see why there should be a recommendation on the minimal API version. I don't consider "it allows modern features" a valid explanation; some client (I don't have any example, that just a supposition) might not want to feel "modern" while still being able to leverage what the OpenSubsonic API has to offer. This sentence is misleading and might pose itself as an entry constraint for people such as myself that would understand it wrong.

**Reply by Tolriq** [reply]
Well the main issue is again how bad the API is built in Subsonic :( Some fields added on later versions are marked as `required` so this makes the OpenSubsonic a mess to handle all the cases about those fields. (Typically some starred/changed/added that are important)
When we will define the new fields for Albums/Artists and songs this will causes issues.

About modern I don't understand what you assume with that term, I guess same as should some personal feelings that are not present in the word. There's nothing about the UI here, but being able to support ratings for everything is modern ;) Being stuck at 1.12 does not allow that, so it's not modern.
